saleor/graphql/attribute/utils.py

```diff
@@ -257,17 +259,36 @@ def _pre_save_reference_values(
             attribute.entity_type  # type: ignore
         ]
         get_or_create = attribute.values.get_or_create
-        return tuple(
-            get_or_create(
-                attribute=attribute,
-                slug=slugify(
-                    f"{instance.id}_{reference.id}",  # type: ignore
-                    allow_unicode=True,
-                ),
-                defaults={"name": getattr(reference, field_name)},
-            )[0]
-            for reference in attr_values.references
-        )
+
+        reference_list = []
+        for ref in attr_values.references:
+            reference_page = None
+            reference_product = None
+
+            if attribute.entity_type == AttributeEntityType.PAGE:
+                if isinstance(ref, Page):
+                    reference_page = ref
+                else:
+                    raise GraphQLError
```

Comment:
Is this GraphQLError handled anywhere in the mutation? I think the validation should happen earlier at the "cleaning" step and not here in the "saving" step. We have for example `_validate_references` which might be a good place to move this validation.